node/src/components/block_proposer.rs

```diff
@@ -384,16 +361,11 @@ impl BlockProposerReady {
             trace!(%hash, "expired deploy rejected from the buffer");
             return;
         }
-        if self.unhandled_finalized.remove(hash.deploy_hash()) {
```

Comment:
I don't really understand why `undhandled_finalized` was remove as part of this change. Can you explain why we don't have that collection anymore?

Reply:
We needed that collection to track finalized deploy hashes for which we were still missing the deploy itself: We couldn't add them to `finalized_deploys` yet, because that needed the header.

Now that we're only tracking the _hashes_ anyway, we can add them to `self.sets.finalized_deploys` right away.

Reply:
This makes sense. Thanks.